### ISSUE TITLE
Apply should throw when path has out of bounds index (was: Composition and inserts in skipped position)

### ISSUE THREAD

**Issue by antoinelyset** [ottypes/json1#22]
Hi,

Again thanks for this great library.
While testing the library with composition and inserts, I stumbled on this behavior.

Is this intended?

```javascript
json1.type.apply(
  [0],
  json1.type.compose(
    json1.insertOp([2], { key: "value" }),
    json1.type.compose(
      json1.insertOp([2, "anotherKey"], "anotherValue"),
      json1.insertOp([2, "lastKey"], "lastValue")
    )
  )
);
=> 
[
  0,
  { key: 'value' },
  { key: 'value', anotherKey: 'anotherValue', lastKey: 'lastValue' }
]
```

Having `{ key: 'value' }` at position 1 is creating a lot of confusion.

**Comment by josephg** [comment]
Ideally that should throw an exception. `2` is an invalid index for the target of your apply() call (`[0]`). The only two valid indexes in this case are 0 (the start of the list) or 1 (the end). Thats a case of garbage in -> garbage out.

If instead you call: `json1.type.apply( [0,1], ...)` then the result is (as expected):

```
[
  0,
  1,
  { key: 'value', anotherKey: 'anotherValue', lastKey: 'lastValue' }
]
```

**Comment by antoinelyset** [comment]
Thanks for the quick answer. Raising and error would help us debug where we emit this kind of incorrect op. I may try to add it today.

**Comment by antoinelyset** [comment]
I think we can close this one. Could you make a NPM version of it :)

**Comment by josephg** [comment]
Sure - 📦 published as ot-json1@1.0.1! Much appreciated!
